### DISCUSSION TITLE
Cross-compiling for linux arm32 on macOS

### DISCUSSION THREAD

**Discussion by imavroukakis** [async-profiler/async-profiler#920] [Q&A] [answered]
Hello! Would there happen to be any instructions anywhere on how to built arm32 binaries on macOS? Additionally, the README states that there are "ports" but I can't seem to find any via the usual means?

Thanks!

**Comment by apangin** [comment, marked as answer]
There is nothing specific to async-profiler in this regard. You may use whatever cross-compiling tools are available. I personally never cross-compiled on macOS; I typically use ARM32 hardware or ARM32 toolchain on Linux-AArch64.

To compile directly on the target architecture, just run `make`
To cross-compile, run `make ARCH=arm CC=/path/to/gcc CXX=/path/to/g++`

"Available ports" mean async-profiler has been ported to the specified platforms. It is known to build and run on these platforms. However, pre-built binaries are provided only for the officially maintained platforms, which currently include x86_64 and aarch64, both on Linux and macOS.

**Comment by imavroukakis** [comment]
Hello @apangin and thanks for the reply. I guess what I was hinting at is, that it would have been nice if there was a little blurb on the README on how to cross-compile binaries, as consumers of your wonderful profiler might not be in-the-know or have access to said hardware 😁

**Reply by apangin** [reply]
Noted. Thanks.
